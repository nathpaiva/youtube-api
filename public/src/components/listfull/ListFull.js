import React, { Component } from 'react';
import { connect } from 'react-redux';

import { loadVideosApi, selectVideo, loadMoreVideosApi } from '../../logics/youtubeApi';

import Highlight from '../highlight/Highlight';
import List from '../list/List';
import Modal from '../modal/Modal';

import allVideos from './listfull.scss'

class ListFull extends Component {
  state = {
    totalVideos: 20,
    isActive: false,
    isFetching: false,
  };

  checkParameter() {
    let term = '';
    if (!!this.props.match.params.q) {
      term = this.props.match.params.q;
    }
    return term;
  }

  componentDidMount() {
    this.props.list({ term: this.checkParameter(), maxResults: this.state.totalVideos });
  }

  componentWillReceiveProps(props) {
    if (props.match.params.q !== this.props.match.params.q) {
      this.props.list({ term: props.match.params.q, maxResults: this.state.totalVideos });
    }
  }

  render() {
    let titleList = !this.props.match.params.q ? "Todos os v√≠deos do Canal" : `Resultados para: "${this.props.match.params.q}"`
    return (
      <div className={allVideos.container_allvideos}>
        <div className={allVideos.allvideos}>
          <List
            titleList={titleList}
            videos={this.props.videos}
            videoSelect={video => {
              this.setState({ isActive: true });
              this.props.changeVideo(video);
            }}
            moreVideos={() => {
              let newTotal = this.state.totalVideos + 10;
              newTotal = newTotal > 50 ? 50 : newTotal
              this.setState({ totalVideos: newTotal, isFetching: true });
              this.props.list({ term: '', maxResults: newTotal });
            }}
            classBtn={'list__more'}
            classList='allvideos'
          />
        </div>
        {this.props.selectedVideo.map(video => <Modal key={video.id.videoId} close={() => {
          this.setState({ isActive: false });
        }} isActive={this.state.isActive} selectedVideo={video} />)}
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    list: (opt) => {
      dispatch(loadVideosApi(opt));
    },
    changeVideo: (video) => {
      dispatch(selectVideo(video));
    }
  }
}

const mapStateToProps = state => {
  return {
    videos: state.listVideosReducer,
    selectedVideo: state.selectVideoReducer,
  }
}

const ListFullRedux = connect(mapStateToProps, mapDispatchToProps)(ListFull);

export default ListFullRedux;
