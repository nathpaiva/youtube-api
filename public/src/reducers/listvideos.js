'user strict';

import { List } from 'immutable';

export function listVideosReducer(state = new List(), action) {

  if (action.type === 'LIST') {
    return new List(action.list);
  }

  return state;
}
