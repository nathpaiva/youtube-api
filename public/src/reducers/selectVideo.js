'user strict';

import { List } from 'immutable';

export function selectVideoReducer(state = new List(), action) {

  if (action.type === 'SELECTFIRST') {
    return [action.list[0]];
  }

  if (action.type === 'SELECTONE') {
    return [action.list];
  }


  return state;
}
