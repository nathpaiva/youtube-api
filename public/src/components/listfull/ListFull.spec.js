'use stict';

import React from 'react';
import { Route, Link, MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { listVideosReducer } from '../../reducers/listvideos';
import { selectVideoReducer } from '../../reducers/selectVideo';

const reducers = combineReducers({ listVideosReducer, selectVideoReducer })
const store = createStore(reducers, applyMiddleware(thunkMiddleware));

import ListFull from './ListFull';


describe('#Test ListFull Component', () => {
  it('#Should ListFull to mach snapshot - all videos without filter', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/videos']}>
        <Route exact path='/videos/:q?' render={routeProps => <ListFull {...routeProps} store={store} />} />
      </MemoryRouter >
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('#Should ListFull to mach snapshot - all videos with filter', () => {
    const videoSelect = jest.genMockFn();

    const tree = renderer.create(
      <MemoryRouter initialEntries={['/videos/ux']}>
        <Route exact path='/videos/:q?' render={routeProps => <ListFull {...routeProps} store={store} />} />
      </MemoryRouter >
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
