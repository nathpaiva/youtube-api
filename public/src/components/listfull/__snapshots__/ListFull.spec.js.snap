// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test ListFull Component #Should ListFull to mach snapshot - all videos with filter 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div>
      <div
        className={undefined}
      >
        <h1>
          Resultados para: "ux"
        </h1>
      </div>
      <div
        className={undefined}
      >
        <ul />
        <button
          className={undefined}
          disabled={undefined}
          onClick={[Function]}
        >
          carregar mais vídeos...
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`#Test ListFull Component #Should ListFull to mach snapshot - all videos without filter 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div>
      <div
        className={undefined}
      >
        <h1>
          Todos os vídeos do Canal
        </h1>
      </div>
      <div
        className={undefined}
      >
        <ul />
        <button
          className={undefined}
          disabled={undefined}
          onClick={[Function]}
        >
          carregar mais vídeos...
        </button>
      </div>
    </div>
  </div>
</div>
`;
